.github/workflows/pull-request-test.yml

```diff
@@ -4,13 +4,14 @@ on: [pull_request]
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    runs-on: macos-latest
```

Comment:
Do we need macos? [Costs 10x](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates) compared to Ubuntu.

Reply:
That is copied from @bohan-amplitude 's change to make the tests more stable.. @bohan-amplitude maybe see if we can make this more stable with ubuntu?

Reply:
let revert this for now.